internal/config/config.go

```diff
@@ -1,70 +1,55 @@
 package config
 
-import "github.com/buildkite/test-engine-client/internal/env"
-
 // Config is the internal representation of the complete test engine client configuration.
 type Config struct {
+	BuildId string `json:"BUILDKITE_BUILD_ID"`
+	JobId   string `json:"BUILDKITE_JOB_ID"`
+	StepId  string `json:"BUILDKITE_STEP_ID"`
 	// AccessToken is the access token for the API.
-	AccessToken string
+	AccessToken string `json:"-"`
 	// Identifier is the identifier of the build.
-	Identifier string
+	Identifier string `json:"BUILDKITE_TEST_ENGINE_IDENTIFIER"`
 	// MaxRetries is the maximum number of retries for a failed test.
-	MaxRetries int
+	MaxRetries int `json:"BUILDKITE_TEST_ENGINE_RETRY_COUNT"`
 	// RetryCommand is the command to run the retry tests.
-	RetryCommand string
+	RetryCommand string `json:"BUILDKITE_TEST_ENGINE_RETRY_CMD"`
 	// Node index is index of the current node.
-	NodeIndex int
+	NodeIndex int `json:"BUILDKITE_PARALLEL_JOB"`
 	// OrganizationSlug is the slug of the organization.
-	OrganizationSlug string
+	OrganizationSlug string `json:"BUILDKITE_ORGANIZATION_SLUG"`
 	// Parallelism is the number of parallel tasks to run.
-	Parallelism int
+	Parallelism int `json:"BUILDKITE_PARALLEL_JOB_COUNT"`
 	// Maximum parallelism when calculating parallelism dynamically.
-	MaxParallelism int
+	MaxParallelism int `json:"-"`
 	// The path to the result file.
-	ResultPath string
+	ResultPath string `json:"-"`
```

Comment:
I've maintained the list of config fields that are excluded from the environment metadata from before this change, but I expect some of these are not intentionally excluded. I may follow up to ensure all fields we might be interested in are sent.